### PR TITLE
feat: Expose ability to hide "Back to sign in" link for all MFA views

### PR DESCRIPTION
**Pull request by jmelberg-okta**
## Description:

- When the configuration option `features.hideSignOutLinkInMFA` is provided views should **not contain** the "Back to sign in" text.

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Added e2e tests
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Reviewers:

- @mauriciocastillosilva-okta

### Issue:

- [OKTA-395411](https://oktainc.atlassian.net/browse/OKTA-395411)

### CODE REVIEW

**Comment by pradeepdewda-okta** [comment]
you can change the title that says `Expose ability to hide "Back to sign in" link for all mfa views`

File: src/v2/models/AppState.js [hunk 1]
```diff
@@ -194,18 +194,15 @@ export default Model.extend({
       transformedResponse.currentAuthenticator?.poll?.refresh;
   },
 
-  // Sign Out link will be displayed in the footer of a form, unless
-  // - widget config hideSignOutLinkInMFA=true or mfaOnlyFlow=true
-  // and form is for identity verification (FORMS_FOR_VERIFICATION)
+  // Sign Out link will be displayed in the footer of a form, unless:
```
Comment (pradeepdewda-okta) [inline src/v2/models/AppState.js:197]: it's unfortunate that the name of the parameter has `Sign Out` in it and now we call it `Back to sign in`.. can we at least mention that in comments and probably change the variable names ?

**Review by pradeepdewda-okta — APPROVED** [review]

File: src/v2/models/AppState.js (hunk 1 above)
Comment (jmelberg-okta) [reply src/v2/models/AppState.js:197]: Agree - we should come up with a deprecation strategy for fields that don't map the best with OIE enabled. I'll create a follow-up for a discussion on this.

Comment (pradeepdewda-okta) [reply src/v2/models/AppState.js:197]: here in comment we can change to `Back to sign in formerly Sign Out link will be displayed .....`

File: test/testcafe/spec/SelectAuthenticatorForVerification_spec.js [hunk 2]
```diff
@@ -209,7 +210,21 @@ test.requestHooks(mockChallengePassword)('should load select authenticator list'
...
+  const selectFactorPage = await setup(t);
+  await renderWidget({
+    features: { hideSignOutLinkInMFA: true },
+  });
+  await t.expect(selectFactorPage.getFormTitle()).eql('Verify it\'s you with an authenticator');
+  // signout link is not visible
+  await t.expect(await selectFactorPage.signoutLinkExists()).notOk();
+});
+
+test.requestHooks(mockChallengePassword)('should load select authenticator list', async t => {
```
Comment (mauriciocastillosilva-okta) [inline test/testcafe/spec/SelectAuthenticatorForVerification_spec.js:225]: do we need this test? Seems it is just testing the title, and I think we have that already in tests above

File: test/unit/spec/v2/models/AppState_spec.js [hunk 3]
```diff
@@ -25,21 +25,17 @@ describe('v2/models/AppState', function() {
       this.initAppState({ idx: { actions: { cancel: 'invalid' } } });
       expect(this.appState.shouldShowSignOutLinkInCurrentForm()).toBe(false);
     });
-    it('returns false if the currentFormName is in FORMS_WITHOUT_SIGNOUT', () => {
-      this.initAppState({ idx: { actions: { cancel: () => {} } } }, FORMS_WITHOUT_SIGNOUT[0]);
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/v2/models/AppState_spec.js:29]: we still need a test for `FORMS_WITHOUT_SIGNOUT`, can we add it back? The test for `FORMS_FOR_VERIFICATION` is that one that is no longer needed

File: test/testcafe/spec/SelectAuthenticatorForVerification_spec.js (hunk 2 above)
Comment (jmelberg-okta) [reply test/testcafe/spec/SelectAuthenticatorForVerification_spec.js:225]: I think this was a copy + paste error.. it shouldn't be here! Will remove. 😞

File: test/unit/spec/v2/models/AppState_spec.js (hunk 3 above)
Comment (jmelberg-okta) [reply test/unit/spec/v2/models/AppState_spec.js:29]: Woah - good catch. I think I got too aggressive with my code deletion and missed this when reviewing. Thank you!

File: test/unit/spec/v2/models/AppState_spec.js [hunk 4]
```diff
@@ -25,21 +25,21 @@ describe('v2/models/AppState', function() {
...
-      expect(this.appState.shouldShowSignOutLinkInCurrentForm()).toBe(false);
+    it('returns false if param hideSignOutLinkInMFA is true but there is no remediation action', () => {
```
Comment (mauriciocastillosilva-okta) [inline test/unit/spec/v2/models/AppState_spec.js:28]: this is a double criteria to return false, right? hideSignOutLinkInMFA=true and no actions.
Not sure if that was the intention, or if we should have one more for hideSignOutLinkInMFA=false and no actions

**Review by mauriciocastillosilva-okta — APPROVED** [review]
looks good, just left one comment I missed in the previous round. 
🐑

File: test/unit/spec/v2/models/AppState_spec.js (hunk 4 above)
Comment (jmelberg-okta) [reply test/unit/spec/v2/models/AppState_spec.js:28]: This should actually be `false`, as we'd never execute the check for remediation actions unless `hideSignOutLinkInMFA` is set to `false`. Updating!

**Pull request closed without merging**